File: src/b1happynumber/JS/happyNumberHelper.js
// happyNumberHelper.js

// Helper function that calculates the sum of squared digits.
function sumOfSquaredDigits(number) {
    let totalSum = 0;

    while (number > 0) {
        const digit = number % 10; // Extract the digit before division
        number = Math.floor(number / 10); // Update the number
        totalSum += digit * digit;
    }

    return totalSum;
}

// Function to check if a number is a happy number
function isHappyNumber(n) {
    let slow = n;
    let fast = sumOfSquaredDigits(n);

    while (fast !== 1 && slow !== fast) {
        slow = sumOfSquaredDigits(slow);
        fast = sumOfSquaredDigits(sumOfSquaredDigits(fast));
    }

    return fast === 1;
}

module.exports = { sumOfSquaredDigits, isHappyNumber };

File: src/b1happynumber/Go/happyNumberHelper.go
// happyNumberHelper.go

package main

// Helper function that calculates the sum of squared digits.
func sumOfSquaredDigits(number int) int {
	totalSum := 0

	for number > 0 {
		digit := number % 10 // Extract the digit before division
		number = number / 10 // Update the number
		totalSum += digit * digit
	}

	return totalSum
}

// Function to check if a number is a happy number
func isHappyNumber(n int) bool {
	slow := n
	fast := sumOfSquaredDigits(n)

	for fast != 1 && slow != fast {
		slow = sumOfSquaredDigits(slow)
		fast = sumOfSquaredDigits(sumOfSquaredDigits(fast))
	}

	return fast == 1
}
